kasa/smart/smartdevice.py

```diff
@@ -195,8 +208,8 @@ async def _initialize_modules(self):
                     mod.__name__,
                 )
                 module = mod(self, mod.REQUIRED_COMPONENT)
-                if module.name not in modules and await module._check_supported():
-                    modules[module.name] = module
+                if await module._check_supported():
```

Comment:
This changes the existing logic and overrides modules if they already exist. If that's wanted, it'd still be a good idea to add a debug log entry for that.

Reply:
[Prior to the ks240 PR](https://github.com/python-kasa/python-kasa/pull/839/files#diff-73c0cd0bf7001c25acb68d47ef1312c9fce12d7114bc23f342b495b421380bd7L174-R199) there wasn't a check to see if the module already existed.  This reverts to that behaviour.